### ISSUE TITLE
sharepointplus with sp-rest-proxy

### ISSUE THREAD

**Issue by mamoussa405** [Aymkdn/SharepointPlus#179]
Hello,
I want to use sharepointplus with sp-rest-proxy to avoid deploying the app whenever i want to test something, i used proxy('http://localhost:8080'), http://localhost:8080 is the path where sp-rest-proxy is runing but it's not working, any ideas how to use sharepointplus whit sp-rest-proxy??, thanks in advance.

**Comment by Aymkdn** [comment]
Sorry I don't know… I'm not using `sp-rest-proxy`.

My setup: I have a local Webpack server that runs, and based on a cookie that is activated on the site, the JS files that are loaded either from SharePoint (remote) or from my https://localhost. This way, by activating the cookie, I can load local files or remote files.

**Comment by mamoussa405** [comment]
> 

yes that's the problem how you can tell sharepointplus to request {tenant name}.sharepoint.com instead of localhost:port??

**Comment by Aymkdn** [comment]
So, `sp-rest-proxy` won't probably work as it seems to be for REST API :)

My setup permits to load the SharePoint page from the SharePoint servers directly.
So if I call https://tenant.sharepoint.com/sites/itdevhr/tmsdev/SitePages/MyTool.aspx with a cookie that says I'm doing local development, then the page will load https://localhost:5000/index.js instead of loading https://tenant.sharepoint.com/sites/itdevhr/tmsdev/SitePages/js/index.ab254d7e82f8.js

This way SharePointPlus can work because it's directly running inside/from the SharePoint server. So it recognizes the base URL, and the authentication is the regular one.

However if you run your own webserver, SharePointPlus won't be able to find the base URL, and might need to use a different authentication to connect to the remote SharePoint server.

**LONG STORY SHORT**: you can try to pass the URL and see if it works…
For example, instead of doing:
```js
$SP().list("My List").get(…)
```
Try with:
```js
$SP().list("My List", "https://{tenant name}.sharepoint.com/sites/itdevhr/tmsdev/").get(…)
```

**Comment by koltyakov** [comment]
sp-rest-proxy supports all SP APIs including legacy SOAP and CSOM, or raw resources.

Let's assume you start your dev workbench at http://localhost:3000

There are two setups:

- sp-rest-proxy is configured as Webpack Dev server middleware (recommended) or local server proxies unauthenticated requests to sp-rest-proxy started on a separate port, then FE should send all the requests to http://localhost:3000/...

- sp-rest-proxy is started on a separate port/process, e.g. http://localhost:3001 and FE sends API request to http://localhost:3001/...

You should never try sending request to an actual SharePoint host as all they will end up unauthenticated.

Depending on a scenario you need routing client requests to end up to proxy. E.g. if a localhost then proxy base path. With [Webpack dev server approach](https://github.com/koltyakov/sp-rest-proxy#webpack-dev-server) you skip this configuration step and always use the same host as application has.

**Comment by mamoussa405** [comment]
@Aymkdn i changed the base url to {tenant name}.sharepoint.com, when i check the network on dev tools it changes but the request failed and i get this error:
`<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot POST /_vti_bin/Webs.asmx</pre>
</body>
</html>`

**Comment by mamoussa405** [comment]
@koltyakov yes it seems that sending request directly won't work, so either deploying the app or using a server proxy, but it seems that it's not possible to configure sharepointplus to work with sp-rest-proxy, so i don't know what is the solution here because i have i project that i have o work on and it's already started using sharepointplus.

**Comment by mamoussa405** [comment]
@Aymkdn @koltyakov i found a solution to do that and it's working, i started sp-rest-proxy on port 8080, and i changed the base url in sharepointplus to http://localhost:8080.
so instead of using:
`$SP.list('list name').get()`
i used:
'$SP.list('list name', 'http://localhost:8080').get()...

**Comment by Aymkdn** [comment]
Try to set `window._spPageContextInfo.webServerRelativeUrl` in your dev environment because it's what SharepointPlus will use on the server to find the url to use. It would be better than hard coding it.

**Comment by mamoussa405** [comment]
okay thank you so much @Aymkdn for your time.
